src/cls/IPM/Main.cls

```diff
@@ -2867,6 +2916,44 @@ ClassMethod EnableIPM(ByRef pCommandInfo)
 	}
 }
 
+ClassMethod History(ByRef pCommandInfo)
+{
+	Set action = $Get(pCommandInfo("parameters", "action"))
+
+	If (action = "find") || (action = "") {
+		Set globally = $$$HasModifier(pCommandInfo, "globally")
+		Set ascend = $$$HasModifier(pCommandInfo, "ascend")
+		Set limit = $$$GetModifier(pCommandInfo, "limit")
+
+		Merge filter = pCommandInfo("data", "zpm")
+		Set mthd = $Select(globally: "GetHistoryGlobally", 1: "GetHistory")
+		Set rs = $Classmethod("%IPM.General.History", mthd, .filter, ascend, limit)
+		Set mthd = $Select(globally: "DisplayArray", 1: "DisplayQueryResult")
+		Do $Classmethod("%IPM.General.History", mthd, rs)
+	} ElseIf action = "details" {
+		Set id = $Get(pCommandInfo("parameters", "argument"))
+		Do ##class(%IPM.General.History).DisplayOneRecord(id)
+	} ElseIf action = "schema" {
+		$$$ThrowOnError($System.SQL.Schema.GetAllColumns("%IPM_General.History", , .columns))
+		Write "Columns available for filtering are: "
+		Set idx = ""
+		For {
+			Set idx = $Order(columns(idx), 1, col)
+			If idx = "" {
+				Quit
+			}
+			If $Extract(col, 1, 3) = "x__" {
+				Continue
+			}
+			Write !, "- ", col
+		}
+	} ElseIf action = "delete" {
+		Do ##class(%IPM.General.History).%KillExtent()
```

Comment:
For delete, should be able to filter what deletes. Such as by time or at a minimum by specific namespace. Should require passing another parameter/modifier to indicate what the scope of delete is